### ISSUE TITLE
Filling of geoshapes in clipped projections

### ISSUE THREAD

**Issue by jwoLondon** [vega/vega#1161]
_The following problem was detected using Vega-Lite, but I think it is probably a Vega issue_

Rendering a filled geoshape that intersects with a geoclipped boundary is not filled correctly:

<img width="358" alt="screen shot 2018-02-16 at 12 24 16" src="https://user-images.githubusercontent.com/1846999/36307508-6f187560-1314-11e8-8820-0cfcafd86fcc.png">

If I use d3 with exactly the same input files, the filling is correct:

<img width="388" alt="screen shot 2018-02-16 at 12 24 43" src="https://user-images.githubusercontent.com/1846999/36307533-926d117e-1314-11e8-94da-4edbea0d55de.png">

Here is a spec which reproduces the problem:

```javascript
{
  "$schema": "https://vega.github.io/schema/vega/v3.0.json",
  "autosize": "pad",
  "padding": 5,
  "width": 400,
  "height": 350,
  "style": "cell",
  "data": [
    {
      "name": "source_1",
      "url": "https://gist.githubusercontent.com/jwoLondon/b971ea2f2ce2a130075fb0bba2d8be02/raw/81cc2b822702b5a58f08798d78504aaec474407d/paris30.json",
      "format": {"type": "topojson","mesh": "parisCircle"}
    },
    {
      "name": "source_0",
      "url": "https://gist.githubusercontent.com/jwoLondon/b971ea2f2ce2a130075fb0bba2d8be02/raw/81cc2b822702b5a58f08798d78504aaec474407d/world-110m.json",
      "format": {"type": "topojson","feature": "countries1"}
    }
  ],
  "projections": [
    {
      "name": "projection",
      "size": {"signal": "[width, height]"},
      "fit": {"signal": "[data('source_0'), data('source_1')]"},
      "type": "gnomonic"
    }
  ],
  "marks": [
    {
      "name": "layer_0_marks",
      "type": "shape",
      "style": ["geoshape"],
      "from": {"data": "source_0"},
      "encode": {
        "update": {
          "fill": {"value": "black"},
          "fillOpacity": {"value": 0.1},
          "stroke": {"value": "white"},
          "strokeWidth": {"value": 0.5}
        }
      },
      "transform": [{"type": "geoshape","projection": "projection"}]
    },
    {
      "name": "layer_1_marks",
      "type": "shape",
      "style": ["geoshape"],
      "from": {"data": "source_1"},
      "encode": {
        "update": {
          "fill": {"value": "#00a2f3"},
          "fillOpacity": {"value": 0.3},
          "stroke": {"value": "#00a2f3"}
        }
      },
      "transform": [{"type": "geoshape","projection": "projection"}]
    }
  ],
  "config": {"axisY": {"minExtent": 30}}
}
```

**Comment by jheer** [comment]
Thanks! I'm able to replicate the result in Vega, but when I try with D3 I'm still seeing the same error -- but using the GeoJSON unpacked by Vega.

I'm wondering if the issue might actually reside at the data loading / TopoJSON level. I should be using the more recent TopoJSON version locally, but I will double check. In the meantime, would you mind sharing your TopoJSON + D3 code so I can replicate your results?

**Comment by jwoLondon** [comment]
Thanks. I've created an [observable notebook](https://beta.observablehq.com/@jwolondon/small-circles) that should allow this to be explored (see '3. Circle loaded from topojson file').

**Comment by jwoLondon** [comment]
And for information the topojson small circle file in the referenced URL, I generated with Mike's `ndjson-cli` tools:

```
echo "{}"  \
| ndjson-map -r d3 'd3.geoCircle().radius(30).center([2.35,48.86])()'  \
| geo2topo parisCircle="-"  \
> data/paris30.json
```

**Comment by jheer** [comment]
Thanks! Using the notebook I figured out the issues involved:
- The D3 example uses `topojson.feature` whereas the Vega example uses `"mesh": "parisCircle"`. If you modify the D3 example to use `topojson.mesh`, the same clipping behavior arises. If this is an error, an issue should probably be filed with the TopoJSON project.
- Modifying the Vega example to use `"feature": "parisCircle"` resulted in a loading error. I have updated vega-loader to fix this issue. The fix has been released in vega-loader@2.0.4 and will be included in the upcoming Vega 3.1 release.
- Also, your D3 example uses TopoJSON v1, where as Vega uses TopoJSON (client) v3. This discrepancy did not affect this particular issue, though.
